File: CS-Examples/07_Conversion/CSVToDataTable.vb
Imports Spire.Xls

Namespace CSVToDataTable

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new instance of Workbook
            Dim workbook As New Workbook()

            ' Load the CSV file from the specified path with the specified delimiter (",")
            workbook.LoadFromFile("..\..\..\..\..\..\Data\CSVSample.csv", ",")

            ' Get the first worksheet from the workbook
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            ' Export the data from the worksheet to a DataTable
            Dim table As DataTable = worksheet.ExportDataTable()

            'Show in data grid
            Me.dataGridView1.DataSource = table
        End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace

File: CS-Examples/07_Conversion/CSVToDataTable.cs
using System;
using System.Windows.Forms;
using Spire.Xls;

namespace CSVToDataTable
{

	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, EventArgs e)
		{
            //Create a workbook
			Workbook workbook = new Workbook();

            //Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\CSVSample.csv", ",");

            //Get the first worksheet
            Worksheet worksheet = workbook.Worksheets[0];

            //Export to datatable
            System.Data.DataTable dataTable = worksheet.ExportDataTable();

            //Show in data grid
            this.dataGridView1.DataSource = dataTable;

            // Dispose of the workbook object to release resources
            workbook.Dispose();
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}
